atlp-rwanda/rca-phantom-team1-bn
# chore: added roles

## What does this PR do
- It implements the roles feature
- It adds the feature tests

## Description
- Admin is able to perform the CRUD operations on roles
- Roles can be seeded at once for development purposes by running `npm run db:seeds`

## How to manually test this 
- Check in the branch `chore-roles`
- Run the server
- Open the Swagger API documentation to test the roles

## Background context
- The Create, Update, and Delete endpoints are protected so they can only be accessed by the admin

## Relevant Trello story 
[9-create-roles-and-set-permissions](https://trello.com/c/a6wcZ3PE/9-create-roles-and-set-permissions)

## Screenshots
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/54368821/226496960-ccaa45af-0032-4326-b92f-063e652d8d79.png">
<img width="1363" alt="image" src="https://user-images.githubusercontent.com/54368821/226497022-27c50446-6531-470f-8856-c83acb5bea37.png">
